assets/javascript/spinner/init.js

```diff
@@ -0,0 +1,177 @@
+class Spinner {
+  container;
+  content;
+  domRequirementsMet;
+  state;
+  timers;
+  config = {
+    apiUrl: "/app-vc-receipt-status",
+    msBeforeInformingOfLongWait: 5000,
+    msBeforeAbort: 25000,
+    msBetweenRequests: 1000,
+  };
+
+  reflectCompletion = () => {
+    this.state.spinnerState = "complete";
+    this.state.done = true;
+  };
+
+  reflectError = () => {
+    window.location.href = "/ipv/page/pyi-technical";
+  };
+
+  reflectLongWait = () => {
+    this.state.spinnerState = "longWait";
+  };
```

Comment:
```suggestion
  reflectCompletion = () => {
    this.spinnerState = "complete";
  };

  reflectError = () => {
    window.location.href = "/ipv/page/pyi-technical";
  };

  reflectLongWait = () => {
    this.spinnerState = "longWait";
  };
```

We don't need `this.state.done` as we can reference the `spinnerState` instead